commit a80f72fb1db205745685804b8c4c153b8dc8392a (refs/changes/50/2950/8)
Author: Brian Bennett <brian.bennett@joyent.com>
Date:   2017-11-16T16:08:27-08:00 (1 year, 11 months ago)
    
    joyent/manta-thoth#153 Want build release
    Reviewed by: Bryan Cantrill <bryan@joyent.com>
    Approved by: Bryan Cantrill <bryan@joyent.com>

diff --git a/.gitignore b/.gitignore
index c2658d7..48ca58b 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1 +1,4 @@
 node_modules/
+opt/
+thoth-*.tar.gz
+node-v*
diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..41021fd
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,20 @@
+UNAME=$(shell uname -s | tr "[:upper:]" "[:lower:]")
+VER=$(shell json -f package.json version)
+
+node_modules: package.json
+	npm install --production
+	@touch node_modules
+
+.PHONY: release github-release
+release: thoth-$(UNAME)-$(VER).tar.gz
+
+publish: release
+	./tools/publish.sh "$(VER)"
+
+thoth-$(UNAME)-$(VER).tar.gz: node_modules
+	./tools/build-release.sh
+
+clean:
+	rm -rf node_modules opt
+	rm -f thoth-$(UNAME)-$(VER).tar.gz
+	rm -rf node-v*-$(UNAME)-*
diff --git a/tools/build-release.sh b/tools/build-release.sh
new file mode 100755
index 0000000..8abe455
--- /dev/null
+++ b/tools/build-release.sh
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+if [[ -n "$TRACE" ]]; then
+    export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+    set -o xtrace
+fi
+
+set -o errexit
+
+node_ver='0.10.38'
+thoth_ver=$(json -f package.json version)
+uname=$(uname -s | tr "[:upper:]" "[:lower:]")
+case "$uname" in
+    darwin) node_arch='x64';;
+    sunos) node_arch='x86';;
+    *) printf 'Sorry, not ready for you'; exit 1;;
+esac
+tar="thoth-${uname}-${thoth_ver}.tar.gz"
+node_dir="node-v${node_ver}-${uname}-${node_arch}"
+node_tar="${node_dir}.tar.gz"
+node_location="https://nodejs.org/download/release/v${node_ver}"
+
+curl -#LOC - "${node_location}/${node_tar}"
+tar zxf "$node_tar"
+
+mkdir -p opt/custom/thoth/{bin,lib}
+cp "${node_dir}/bin/node" opt/custom/thoth/bin
+if [[ $uname == sunos ]]; then
+    cp /opt/local/gcc47/lib/libstdc++.so.6 opt/custom/thoth/lib
+    cp /opt/local/gcc47/lib/libgcc_s.so.1 opt/custom/thoth/lib
+fi
+cp -r node_modules opt/custom/thoth
+
+tar zcf "$tar" opt
diff --git a/tools/manta-thoth-build.sh b/tools/manta-thoth-build.sh
new file mode 100755
index 0000000..64bbc0e
--- /dev/null
+++ b/tools/manta-thoth-build.sh
@@ -0,0 +1,11 @@
+#!/bin/bash -x
+
+set -o errexit
+
+job=$(
+mjob create --close -m 'set -o errexit ;
+cd /var/tmp ;
+git clone https://github.com/joyent/manta-thoth ;
+cd manta-thoth ;
+make publish')
+mjob watch "$job"
diff --git a/tools/publish.sh b/tools/publish.sh
new file mode 100755
index 0000000..e4ba56b
--- /dev/null
+++ b/tools/publish.sh
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+ver="$1"
+uname=$(uname -s | tr "[:upper:]" "[:lower:]")
+name="thoth-${uname}-${ver:?}.tar.gz"
+latest="thoth-${uname}-latest.tar.gz"
+
+mput -f "$name" "/thoth/public/$name"
+mln "/thoth/public/$name" "/thoth/public/$latest"
